File: src/main/java/com/jarmison/varejo/online/api/model/Lancamentos.java
package com.jarmison.varejo.online.api.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.jarmison.varejo.online.api.Enums.Cargos;
import com.jarmison.varejo.online.api.Enums.TipoMovimentacao;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Data
@Entity
public class Lancamentos {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long codigolancamento;
    @ManyToOne
    @JoinColumn(name = "produto_id")
    private Produto produto;
    @Enumerated(EnumType.STRING)
    private Cargos cargo;
    @Enumerated(EnumType.ORDINAL)
    private TipoMovimentacao tipoMovimentacao;
    private Integer quantidade;

    private Double saldo;
    @CreationTimestamp
    @Column(nullable = false,updatable = false)
    @JsonFormat(pattern="dd/MM/yyyy")
    private Date dataMovimentacao;
    private String motivoMovimentacao;
    private String documento;


}

File: FRONTEND-VAREJONLINE/src/app/components/models/lancamentos.ts
import { Cargo } from "./cargos"
import { Produto } from "./produtos"
import { TipoMovimentacao } from "./tipo_movimentacao"

export class Lancamentos{
    codigolancamento:number
    produto:Produto
    cargo:Cargo
    tipoMovimentacao:TipoMovimentacao
    quantidade:number
    dataMovimentacao:Date
    documento:string
    motivoMovimentacao:string
    saldo:number
}